control-plane/pkg/contract/subscriptionsapi.go

```diff
@@ -0,0 +1,196 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package contract
+
+import v1 "knative.dev/eventing/pkg/apis/eventing/v1"
+
+// NewExactFilter converts the SubscriptionsAPIFilter into the exact dialect of
+// the DialectedFilter as defined in CloudEvents Subscriptions API.
+//
+// Exact contains exactly one attribute where the key is the name of the CloudEvent
+// attribute and its value is the string value which must exactly match the value
+// of the CloudEvent attribute.
+//
+// See "CNCF CloudEvents Subscriptions API" > "3.2.4.1 Filters Dialects"
+// https://github.com/cloudevents/spec/blob/main/subscriptions/spec.md#all-filter-dialect
+func NewExactFilter(f v1.SubscriptionsAPIFilter) *DialectedFilter {
+	return &DialectedFilter{
+		Filter: &DialectedFilter_Exact{
+			Exact: &Exact{
+				Attributes: f.Exact,
+			},
+		},
+	}
+}
+
+// NewPrefixFilter converts the SubscriptionsAPIFilter into the suffix dialect of
+// the DialectedFilter as defined in CloudEvents Subscriptions API.
+//
+// Prefix contains exactly one attribute where the key is the name of the CloudEvent
+// attribute which value must start with the value specified.
+//
+// See "CNCF CloudEvents Subscriptions API" > "3.2.4.1 Filters Dialects"
+// https://github.com/cloudevents/spec/blob/main/subscriptions/spec.md#prefix-filter-dialect
+func NewPrefixFilter(f v1.SubscriptionsAPIFilter) *DialectedFilter {
+
+	p := &Prefix{}
+	// Eventing Trigger API is conforming to the Subscriptions API, so this map has only a single
+	// key-value pair
+	for k, v := range f.Prefix {
+		p.Attribute = k
+		p.Prefix = v
+		// We only expect a single element, but let's break anyway
+		break
+	}
+
+	return &DialectedFilter{
+		Filter: &DialectedFilter_Prefix{
+			Prefix: p,
+		},
+	}
+}
+
+// NewSuffixFilter converts the SubscriptionsAPIFilter into the suffix dialect of
+// the DialectedFilter as defined in CloudEvents Subscriptions API.
+//
+// Suffix contains exactly one attribute where the key is the name of the CloudEvent
+// attribute which value must end with the value specified.
+//
+// See "CNCF CloudEvents Subscriptions API" > "3.2.4.1 Filters Dialects"
+// https://github.com/cloudevents/spec/blob/main/subscriptions/spec.md#suffix-filter-dialect
+func NewSuffixFilter(f v1.SubscriptionsAPIFilter) *DialectedFilter {
+
+	p := &Suffix{}
+	// Eventing Trigger API is conforming to the Subscriptions API, so this map has only a single
+	// key-value pair
+	for k, v := range f.Suffix {
+		p.Attribute = k
+		p.Suffix = v
+		// We only expect a single element, but let's break anyway
+		break
+	}
+
+	return &DialectedFilter{
+		Filter: &DialectedFilter_Suffix{
+			Suffix: p,
+		},
+	}
+}
```

Comment:
As the other comment on the PR, there shouldn't be a difference between these 3 filters.

Reply:
Fixed